autorest/authorization_storage.go

```diff
@@ -35,6 +35,9 @@ const (
 	// SharedKey is used to authorize against blobs, files and queues services.
 	SharedKey SharedKeyType = "sharedKey"
 
+	// SharedKey is used to authorize against the account.
+	SharedKeyForAccount SharedKeyType = "sharedKeyAccount"
```

Comment:
I think this can be removed, just add content length in the string-to-sign input if it's not equal to zero.